### PR TITLE
Topic/ignore data array

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

**Comment by imikejackson** [comment]
The unit tests pass on macOS. Have not tried the build on Windows or Linux.

File: Source/Plugins/Processing/ProcessingFilters/FillBadData.cpp [hunk 1]
```diff
@@ -439,7 +442,7 @@ void FillBadData::execute()
   notifyStatusMessage(getHumanLabel(), "Complete");
 }
 
-// -----------------------------------------------------------------------------
+// -----------------------------------------------------------------------------4
```
Comment (mmarineBlueQuartz) [inline Source/Plugins/Processing/ProcessingFilters/FillBadData.cpp]: Delete the '4'

File: Source/Plugins/Processing/ProcessingFilters/MinSize.cpp [hunk 2]
```diff
@@ -404,6 +409,14 @@ void MinSize::assign_badpoints()
     }
     QString attrMatName = m_FeatureIdsArrayPath.getAttributeMatrixName();
     QList<QString> voxelArrayNames = m->getAttributeMatrix(attrMatName)->getAttributeArrayNames();
+    for(const auto& dataArrayPath : m_IgnoredDataArrayPaths)
+    {
+      if(voxelArrayNames.contains(dataArrayPath.getDataArrayName()))
+      {
+        voxelArrayNames.removeAll(dataArrayPath.getDataArrayName());
```
Comment (mmarineBlueQuartz) [inline Source/Plugins/Processing/ProcessingFilters/MinSize.cpp]: contains(const T&) needs to iterate over all items at worst to return that it doesn't need to change the list.  There shouldn't be a performance loss if you remove the check and there is no change, but there will be a performance boost if even one item needs to be removed.

Comment (imikejackson) [reply Source/Plugins/Processing/ProcessingFilters/MinSize.cpp]: Good catch. Fixed in latest push.

File: Source/Plugins/Processing/ProcessingFilters/FillBadData.cpp (hunk 1 above)
Comment (imikejackson) [reply Source/Plugins/Processing/ProcessingFilters/FillBadData.cpp]: Fixed.

**Review by mmarineBlueQuartz — APPROVED** [review]
Everything looks good.  std::array initialization would look better if compilers accepted the previous format as per standard.

**Comment by imikejackson** [comment]
Clang is the compiler that warns on single bracket use even though it is perfectly legal C++11 code.

**Pull request merged**